### PR TITLE
moving worker in to triage.python

### PR DESCRIPTION
**Pull request by bryanar**
This exists and is stable, bringing it here for 'fficial reasons.

### CODE REVIEW

**Comment by schaabs** [comment]
One overall comment.  I could forsee having multiple python worker roles in this project.  Could we name config.json and worker.py with more descriptive names

File: src/triage.python/monitor.sh [hunk 1]
```diff
@@ -0,0 +1,6 @@
+#!/bin/bash
+echo "Starting reliability worker." >>./out.txt 2>&1 # pipe both std err and stdout to this file.
+until ./worker.py >>./out.txt 2>&1; do
```
Comment (schaabs) [inline src/triage.python/monitor.sh]: Can you make the log file a descriptive name

**Comment by bryanar** [comment]
@schaabs What kinds of other workers are you suspecting may exist? Linux/Windows should be the same. Do you mean something like 'dumpling_worker.py' maybe?

File: src/triage.python/worker.py [hunk 2]
```diff
@@ -0,0 +1,375 @@
...
+
+from time import sleep
+from datetime import datetime
+from os import path
+
+from azure.servicebus   import ServiceBusService, Message, Queue
+from azure.storage      import CloudStorageAccount
+
+_thisFolder = path.dirname(path.realpath(__file__))
+
```
Comment (schaabs) [inline src/triage.python/worker.py]: encapsulate the root level code into methods

File: src/triage.python/worker.py [hunk 3]
```diff
@@ -0,0 +1,375 @@
...
+        INFORMATIVE(status)
+        continue
+    else:
+        _path_check_succeeded = False
+        FAILURE(status, 4)    
+        
+if not _path_check_succeeded:
+    FAILURE('configured paths need to exist, and must be absolute.', 8)
+
+site.addsitedir(PATH('lldb_py_path'))
```
Comment (schaabs) [inline src/triage.python/worker.py]: Is this code at the root level or is the indentation messed up somehow?  If it's intentional it's a bit confusing

File: src/triage.python/worker.py [hunk 4]
```diff
@@ -0,0 +1,375 @@
...
+
+
+def UpdateWorkerState(state, newState):
+    VERBOSE('Setting worker state to "%s"' % newState)
+    metadata = blob_service.get_container_metadata(state)
+    metadata['worker_state'] = newState;
+    blob_service.set_container_metadata(state, metadata)
+
+# given a list of uris, produce a list of locations of core files.
+def DownZip(uris, state):
```
Comment (schaabs) [inline src/triage.python/worker.py]: Could this be refactored to use the existing downzip.py?

File: src/triage.python/worker.py [hunk 5]
```diff
@@ -0,0 +1,375 @@
...
+    debugger = debuggerTuple[0]
+    interpreter = debuggerTuple[1]
+    target = debuggerTuple[2]
+    process = debuggerTuple[3]
+    pathTuple = debuggerTuple[4]
+
+    # just delete the /home/DotNet folder for now.
+    VERBOSE('cleaning up.')
+    lldb.SBDebugger.Destroy(debugger)
+    shutil.rmtree('/home/DotNetBot/') # This will likely not work in the future when we receive dumps from other people.
```
Comment (schaabs) [inline src/triage.python/worker.py]: We can't be guaranteed that dump zips will unpack to here (as per you're comment.

File: src/triage.python/worker.py (hunk 3 above)
Comment (bryanar) [inline src/triage.python/worker.py]: The idea behind this is that the script will fail up front and with a clear meaning when essential files are missing. I am encapsulating it in a function named CheckEssentialPaths. It's meant to aid in adding additional files too. If we need to add additional dependencies, just write the code and add yourself to this dictionary and all of the logic for properly configuring/checking existence of the file is already taken care of.

Comment (schaabs) [inline src/triage.python/worker.py]: Sounds good.

File: src/triage.python/monitor.sh (hunk 1 above)
Comment (bryanar) [inline src/triage.python/monitor.sh]: Made it analysis_worker_log.log

File: src/triage.python/worker.py (hunk 3 above)
Comment (bryanar) [inline src/triage.python/worker.py]: Yes, I will think about this a bit more.

File: src/triage.python/worker.py (hunk 4 above)
Comment (bryanar) [inline src/triage.python/worker.py]: #13 opened.

File: src/triage.python/worker.py (hunk 5 above)
Comment (bryanar) [inline src/triage.python/worker.py]: #15 opened.

**Comment by bryanar** [comment]
I did my best. It may need some more work but additional issues will have to be opened. I am moving on.

**Pull request merged**